File: Parser (JAVA)/src/main/java/dao/Activity.java
package dao;

import java.util.Map;
/**
Base model class to hold the Activity coming from Firebase 
*/
public class Activity {

    public static String TYPE_TAP = "tap";
    public static String TYPE_SCROLL = "scroll";
    public static String TYPE_DISTRACTED = "distracted";
    public static String TYPE_LOOKING = "looking";
    public static String TYPE_URL_CHANGE = "url_change";

    private String type;
    private long timeStamp;
    private Map<String, Object> metaData;

    public Activity() {

    }

    public Activity(String type, long timeStamp, Map<String, Object> metaData) {
        this.type = type;
        this.timeStamp = timeStamp;
        this.metaData = metaData;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public long getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(long timeStamp) {
        this.timeStamp = timeStamp;
    }

    public Map<String, Object> getMetaData() {
        return metaData;
    }

    public void setMetaData(Map<String, Object> metaData) {
        this.metaData = metaData;
    }

    @Override
    public String toString() {
        return "Activity{" +
                "type='" + type + '\'' +
                ", timeStamp=" + timeStamp +
                ", metaData=" + metaData +
                '}';
    }
}

File: IOS App/Eyes Tracking/Activity.swift
//
//  Activity.swift
//  Eyes Tracking
//
//  Created by Shaheed on 7/28/19.
//  Copyright © 2019 virakri. All rights reserved.
//

import UIKit

class Activity {
    
    // Mark: Enum - Activity Types
    static let TYPE_TAP = "tap"
    static let TYPE_SCROLL = "scroll"
    static let TYPE_DISTRACTED = "distracted"
    static let TYPE_LOOKING = "looking"
    static let TYPE_URL_CHANGE = "url_change"
    
    // MARK: Properties
    var type: String
    var timeStamp: Int
    var metaData: [String: Any]
    
    init(type: String, timeStamp: Int, metaData: [String: Any]) {
        self.type = type
        self.timeStamp = timeStamp
        self.metaData = metaData
    }
    
    var parse: [String: Any] {
        return [
            "type": self.type,
            "timeStamp": self.timeStamp,
            "metaData": self.metaData
        ]
    }
    
    static func toJson(activities: [Activity]) -> [[String: Any]] {
        var json: [[String: Any]] = []
        for (_, activity) in activities.enumerated() {
            json.append(activity.parse)
        }
        return json
    }
    
}
